RubberduckTests/Grammar/SelectionExtensionsTests.cs

```diff
@@ -0,0 +1,436 @@
+﻿using System.Linq;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Rubberduck.Parsing;
+using Rubberduck.Parsing.Grammar;
+using static Rubberduck.Parsing.Grammar.VBAParser;
+using Rubberduck.VBEditor;
+using RubberduckTests.Mocks;
+using Antlr4.Runtime.Misc;
+using System.Collections.Generic;
+
+namespace RubberduckTests.Grammar
+{
+    [TestClass]
+    public class SelectionExtensionsTests
+    {
+        public class CollectorVBAParserBaseVisitor<Result> : VBAParserBaseVisitor<IList<Result>>
+        {
+            private List<Result> defaultResult = new List<Result>();
+
+            protected override IList<Result> DefaultResult => defaultResult;
+            /*
+            protected override IList<Result> AggregateResult(IList<Result> firstResult, IList<Result> secondResult)
+            {
+                if (firstResult != null && secondResult != null)
+                    return firstResult.Concat(secondResult) as IList<Result>;
+
+                if (secondResult == null)
+                    return firstResult;
+
+                return secondResult;
+            }*/
```

Comment:
dead code?

Reply:
According to the documentation, the base implementation of `AggregateResults` always takes the second result. So, if you do not overwrite this, you will always get the result from the last child.

Reply:
Not any more - see my other comment to @MDoerner regarding the visitor.